### ISSUE TITLE
Replacing services failure

### ISSUE THREAD

**Issue by dave08** [argoproj/argo-cd#2150]
**Describe the bug**
When updating a service I got: 
```
kubectl failed exit status 1: Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
The Service "ingress-nginx-nginx-ingress-controller" is invalid: spec.clusterIP: Invalid value: "": field is immutable`
```

It seems like this issue applies to the problem: https://github.com/kubernetes/kubernetes/issues/11237 

ArgoCD version: 1.1.1

**Comment by simster7** [comment]
Is this during an `argocd app sync` or a direct `kubectl apply`?

**Comment by alexec** [comment]
It's straight forward to repo this. I would have expected it to be fixed already. Makes me wonder why have I missed.

**Comment by alexec** [comment]
Note to self-what about `--force` option in CLI or UI?

**Comment by alexec** [comment]
@dave08 - you can update immutable resources using `argocd app sync APPNAME --force`. ATM - this in only available via the CLI, not the UI I'm afraid. I've created a PR to add this to the UI.

**Comment by alexec** [comment]
Fixed in v1.3

**Comment by syst0m** [comment]
Ran into the same issue, but not on updating the service.
Rather it manifests on first sync of the `cluster-autoscaler` helm chart.
Using Argo v1.21.

```
time="2019-09-20T09:10:09Z" level=error msg="`kubectl --kubeconfig /dev/shm/259746214 -f - apply -n argocd` failed exit status 1: The Service \"cluster-autoscaler-aws-cluster-autoscaler\" is invalid: spec.clusterIP: Invalid value: \"\": field is immutable" execID=Y0bd4
time="2019-09-20T09:10:09Z" level=info msg="adding resource result, status: 'SyncFailed', phase: 'Failed', message: 'kubectl failed exit status 1: The Service \"cluster-autoscaler-aws-cluster-autoscaler\" is invalid: spec.clusterIP: Invalid value: \"\": field is immutable'" application=cluster-autoscaler kind=Service name=cluster-autoscaler-aws-cluster-autoscaler namespace=argocd phase=Sync
```
